inc/Engine/Optimization/RUCSS/AbstractAPIClient.php

```diff
@@ -77,14 +79,48 @@ protected function handle_post( array $args ): bool {
 			'wpr_key'   => $this->options->get( 'consumer_key', '' ),
 		];
 
-		$response = wp_remote_post(
-			$api_url . $this->request_path,
-			$args
-		);
+		switch ( strtolower( $type ) ) {
+			case 'get':
+				$response = wp_remote_get(
+					$api_url . $this->request_path,
+					$args
+				);
+				break;
+
+			case 'post':
+			default:
+				$response = wp_remote_post(
+					$api_url . $this->request_path,
+					$args
+				);
+				break;
+		}
```

Comment:
This switch is kind of unnecessary, we can use `wp_remote_request` and add the `METHOD` to the `args` array. That way it will make the method very adaptable.